typings/embedded_network/http/client.d.ts

```diff
@@ -20,12 +20,15 @@
 
 declare module "embedded:network/http/client" {
   import type { Buffer } from "embedded:io/_common"
+  import type { TCPDevice } from "embedded:io/socket/tcp";
+  import type { TLSDevice } from "embedded:io/socket/tls";
+  import type { DNSUDPDevice } from "embedded:io/socket/dns";
 
-  export interface ClientOptions {
-    socket: any
-    port?:number
-    host: string
-    dns: any
+  export interface HTTPClientOptions {
+    socket: TCPDevice | TLSDevice;
+    port?: number;
+    host: string;
+    dns: DNSUDPDevice;
```

Comment:
There's no requirement that this is a DNS using UDP. It could be another resolution transport. But, for the moment this is probably OK?